src/app/clusters/groups-server/groups-server.cpp

```diff
@@ -232,141 +261,150 @@ bool emberAfGroupsClusterViewGroupCallback(app::CommandHandler * commandObj, con
-            GroupId groupId = iter.GetValue();
-            emberAfGroupsClusterPrint(" 0x%02" PRIx16, groupId);
-            for (j = 0; j < EMBER_BINDING_TABLE_SIZE; j++)
+            TLV::TLVType type;
+            ReturnErrorOnFailure(
+                writer.StartContainer(TLV::ContextTag(to_underlying(Commands::GetGroupMembershipResponse::Fields::kGroupList)),
+                                      TLV::kTLVType_Array, type));
             {
-                EmberBindingTableEntry entry;
-                status = emberGetBinding(j, &entry);
-                if ((status == EMBER_SUCCESS) && (entry.type == EMBER_MULTICAST_BINDING))
+                GroupDataProvider::GroupMapping mapping;
+                size_t requestedCount = 0;
+                size_t matchCount     = 0;
+                ReturnErrorOnFailure(mCommandData.groupList.ComputeSize(&requestedCount));
+
+                emberAfGroupsClusterPrint("RX: GetGroupMembership [");
+                if (0 == requestedCount)
                 {
-                    if (entry.local == emberAfCurrentEndpoint() && entry.groupId == groupId)
+                    // 1.3.6.3.1. If the GroupList field is empty, the entity SHALL respond with all group identifiers of which the
+                    // entity is a member.
+                    while (mIterator->Next(mapping))
                     {
-                        list[listLen]     = EMBER_LOW_BYTE(groupId);
-                        list[listLen + 1] = EMBER_HIGH_BYTE(groupId);
-                        listLen           = static_cast<uint8_t>(listLen + 2);
-                        count++;
+                        ReturnErrorOnFailure(app::DataModel::Encode(writer, TLV::AnonymousTag, mapping.group));
+                        matchCount++;
+                        emberAfGroupsClusterPrint(" 0x%02" PRIx16, mapping.group);
                     }
                 }
+                else
+                {
+                    while (mIterator->Next(mapping))
+                    {
+                        auto iter = mCommandData.groupList.begin();
+                        while (iter.Next())
+                        {
+                            if (mapping.group == iter.GetValue())
+                            {
+                                ReturnErrorOnFailure(app::DataModel::Encode(writer, TLV::AnonymousTag, mapping.group));
+                                matchCount++;
+                                emberAfGroupsClusterPrint(" 0x%02" PRIx16, mapping.group);
+                                break;
+                            }
+                        }
+                        ReturnErrorOnFailure(iter.GetStatus());
+                    }
+                }
+                emberAfGroupsClusterPrintln("]");
+
+                // 1.3.6.3.1: If the GroupList field contains one or more group-id’s but does not contain any group
+                // of which the entity is a member, the entity SHALL only respond if the command is unicast
```

Comment:
In Matter, there are no unicast group commands and no unicast responses to multicast group commands. This seems like an issue w/ CSA data model unification where there is a statement in Groups cluster that is fundamentally applicable only to APS and not to Matter.

Reply:
I have removed the condition.